python/bin/watchman-wait-aio

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python3
+
+from pywatchman.aioclient import AIOClient as WatchmanClient
+
+import argparse
+import asyncio
+import pywatchman
+import os
+import sys
+
+def _with_timeout(task, timeout):
+    if timeout > 0:
+        return asyncio.wait_for(task, timeout)
+    return task
+
+
+class Subscription(object):
+    '''A single subscription.'''
+
+    def __init__(self, abspath, name):
+        self.name    = name    # Our name for this subscription
+        self.abspath = abspath # Full path to the dir being watched
+        self.root    = None    # Watched root
+        self.relpath = None    # Path relative to root of thr dir being watched
+
+    def __str__(self):
+        return 'Subscription(name=%s, root=%s, abspath=%s, relpath=%s)' % (
+            self.name, self.root, self.abspath, self.relpath)
+
+    async def activate(self, client, patterns=None, fields=['name']):
+        '''Activate the subscription. This function is a coroutine.'''
+        dir_to_watch = self.abspath
+
+        if patterns:
+            expr = ['anyof']
+            for p in patterns:
+                expr.append(['match', p, 'wholename', {
+                    'includedotfiles': True}])
+        else:
+            expr = ['true']
+
+        if not os.path.isdir(self.abspath):
+            # Need to watch its parent
+            dir_to_watch = os.path.dirname(self.abspath)
+            expr = ['name', os.path.basename(self.abspath)]
+
+        query = {
+            'expression': expr,
+            'fields': fields,
+        }
+
+        watch = await client.query('watch-project', dir_to_watch)
+
+        if 'warning' in watch:
+            print('WARNING: ', watch['warning'], file=sys.stderr)
```

Comment:
If I recall correctly, `watchman-wait` (and, hence, `watchman-wait-aio`) is a stand-alone application. @sid0, @wez is there some library-like use for this program?